recipes/rangesnext/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.12)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```

```suggestion
cmake_minimum_required(VERSION 3.1)
project(test_package)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

find_package(rangesnext REQUIRED CONFIG)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} rangesnext::rangesnext)
```

- Cmake 3.1 is enough for this simple cmake file.
- The `conan_basic_setup(TARGETS)` won't inject defines and include dirs magically when building
- The `find_package(rangesnext REQUIRED CONFIG)` will make Cmake search for rangesnextConfig.cmake, which will be generated by Conan, because `cmake_find_package_multi` is listed as generator
- The `target_link_libraries(${PROJECT_NAME} rangesnext::rangesnext)` target instead of library names. Those targets are in rangesnextConfig.cmake